tests/plenary/popup_spec.lua

```diff
@@ -130,6 +131,148 @@ describe("plenary.popup", function()
+      eq(main_wid, cur_wid)
+      vim.api.nvim_win_close(popup_wid, true)
+    end)
+
+    it("enter = false", function()
+      local main_wid = vim.fn.win_getid()
+      local popup_wid = popup.create("hello there", {
+        enter = false,
+      })
+      cur_wid = vim.fn.win_getid()
+      -- current window should still be the main window
+      eq(main_wid, cur_wid)
+      vim.api.nvim_win_close(popup_wid, true)
+    end)
+
+    it("enter = true", function()
+      local main_wid = vim.fn.win_getid()
+      local popup_wid = popup.create("hello there", {
+        enter = true,
+      })
+      cur_wid = vim.fn.win_getid()
+      -- current window should be the popup
+      eq(popup_wid, cur_wid)
+      vim.api.nvim_win_close(popup_wid, true)
+    end)
+  end)
+
+  describe("moved option", function()
+    local callback_result
+    local function callback(wid, result)
+      callback_result = result
+    end
+
+    it("with moved but not used", function()
+      callback_result = nil
+      local popup_wid = popup.create("hello there", {
+        moved = "any",
+        callback = callback,
+      })
+      vim.api.nvim_win_close(popup_wid, true)
+      eq(-1, callback_result)
+    end)
+
+    --[[
+    local function populate()
+      local wnr = vim.fn.winnr()
+      local bnr = vim.fn.winbufnr(wnr)
+      vim.fn.setbufline(bnr, 1, { "--one--", "--two--", "--three--", "--four--" })
+      vim.fn.cursor(1, 1)
+    end
+    ]]
+
+    --[[
+    it("without moved", function()
+      --async()
+      callback_result = nil
+      populate()
+      local popup_wid = popup.create("hello there", {
+        callback = callback,
+      })
+      -- move the cursor, should not do callback
+      vim.fn.cursor(2, 1)
+      local timer = vim.uv.new_timer()
+      timer:start(10, 0, function()
+        eq(nil, callback_result)
+        vim.api.nvim_win_close(popup_wid, true)
+        --done()
+      end)
+    end)
+    ]]
+
+    --[[
+    it("with moved", function()
+      --async()
+      callback_result = nil
+      populate()
+      local popup_wid = popup.create("hello there", {
+        moved = "any",
+        callback = callback,
+      })
+      -- move the cursor, window closes and callback invoked
+      vim.fn.cursor(2, 1)
+      local done = false
+      vim.defer_fn(function()
+        --eq(-1, callback_result)
+        --eq(0, callback_result)
+        done = true
+      end, 100)
+      vim.wait(100, function()
+        eq(0, callback_result)
+        return done
+      end, 60)
+      --if -1 ~= callback_result then
+      --  -- window wasn't closed
+      --  vim.api.nvim_win_close(popup_wid, true)
+      --end
+
+      --local timer = vim.uv.new_timer()
+      --timer:start(10, 0, function()
+      --  eq(-1, callback_result)
+      --  if -1 ~= callback_result then
+      --    -- window wasn't closed
+      --    vim.api.nvim_win_close(popup_wid, true)
+      --  end
+      --  --done()
+      --end)
+    end)
+    ]]
```

Comment:
If this won't be used in the end, remove it along with the `require("plenary.async").tests.add_to_env()` comment on line 2, but if you want to try and add the async testing functionality, you can head to [async test folder](https://github.com/nvim-lua/plenary.nvim/blob/master/tests/plenary/async/test_spec.lua) and see how it's used there.

Reply:
I took this out and stashed it. I'll reconsider adding it at a later time.